src/compiler/program.ts

```diff
@@ -690,7 +706,11 @@ namespace ts {
         function getCommonSourceDirectory() {
             if (commonSourceDirectory === undefined) {
                 const emittedFiles = filter(files, file => sourceFileMayBeEmitted(file, options, isSourceFileFromExternalLibrary));
-                if (options.rootDir && checkSourceFilesBelongToPath(emittedFiles, options.rootDir)) {
+                if (options.composite) {

```

Comment:
should we check that we have a configFilePath? since if you have niether, normalizeSlashes will get `undefined`.

Reply:
Added an assert